launcher.py

```diff
@@ -0,0 +1,405 @@
+        else:
+            pip_path = venv_dir / "bin" / "pip"
+            if not pip_path.exists():
+                raise FileNotFoundError("pip not found in the bin folder.")
+
+        run_command(f"{str(pip_path)} install -r requirements.txt")
+        print("Python dependencies installed.")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to install Python dependencies: {e}")
+        sys.exit(1)
+
+
+def build_cpp_dependencies():
+    try:
+        print("Building MediaProcessor...")
+        build_dir = os.path.join("MediaProcessor", "build")
+        if not os.path.exists(build_dir):
+            os.makedirs(build_dir)
+
+        run_command("cmake -DCMAKE_BUILD_TYPE=Release ..", cwd=build_dir)
+        run_command("cmake --build . --config Release", cwd=build_dir)
+
+        print("MediaProcessor built successfully.")
+    except subprocess.CalledProcessError as e:
+        print(f"Failed to build MediaProcessor: {e}")
+        sys.exit(1)
+
+
+def update_config(system):
+    try:
+        with open(CONFIG_FILE, "r") as config_file:
+            config = json.load(config_file)
+
+        if system == "Windows":
+            for key, value in config.items():
+                if type(value) == str:
+                    config[key] = value.replace("/", "\\")
+
+        with open(CONFIG_FILE, "w") as config_file:
+            json.dump(config, config_file, indent=4)
+
+    except FileNotFoundError:
+        print(f"{CONFIG_FILE} not found. Please create a default config.json file.")
+        sys.exit(1)
+
+
+def launch_web_application(system):
+    try:
+        if not venv_dir.exists():
+            create_virtualenv()
+
+        if system == "Windows":
+            python_path = str(venv_dir / "Scripts" / "python.exe")
+        else:
+            python_path = str(venv_dir / "bin" / "python")
+
+        # Start the backend
+        app_process = subprocess.Popen(
+            [python_path, "app.py"],
+            stdout=None if current_debug_level >= DebugLevel.APP else subprocess.DEVNULL,
+            stderr=subprocess.PIPE,
+        )
+        atexit.register(app_process.terminate)
+
+        # Give the process some time to initialize
+        time.sleep(0.5)
+
+        # Check if the process is still running
+        if app_process.poll() is not None:
+            error_output = app_process.stderr.read().decode("utf-8")
+            print(f"Error starting the backend: {error_output}")
+            sys.exit(1)
+
+        webbrowser.open("http://127.0.0.1:8080")
+
+        print("Web application running. Press Enter to stop.")
+        input()  # Block until the user presses Enter
+
+    except Exception as e:
+        print(f"An error occurred: {e}")
+        sys.exit(1)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Setup for MediaProcessor Application")
+    parser.add_argument("--mode", choices=["web"], help="Specify mode to launch")
```

Comment:
I'd go with `--app` instead of `--mode` here as we want to launch a specific application.
Also, we'd like to make it easy on the user; so if they simply run this launcher, it should install everything and default to the web app if no arguments were provided.

Reply:
we can use the argument as
`--mode=app` or `--mode=web`

using --app and --web as arguments have problems like what if a user use both --web and --app, Then both web and app will launch and there will be issue with the backend using same port.

but we can go with --app and --web as seperate arguments if you want.

Reply:
Yeah what I meant was like `--app=web` and not `--app` as a standalone arg. Since it's a launcher, we should provide a way to select which app to launch and using the word `mode` feels off to me. 

For the dependency installation and configuration pipeline, we can add something like `--install-dependencies` that would automatically be executed anyway, but would still be nice to provide is to perform an install without launching any app.

Reply:
oh! I get it.